packages/react-aria-components/src/DragAndDrop.tsx

```diff
@@ -75,7 +75,26 @@ export function useDndPersistedKeys(selectionManager: MultipleSelectionManager,
     dropTargetKey = dropState.target.key;
     if (dropState.target.dropPosition === 'after') {
       // Normalize to the "before" drop position since we only render those to the DOM.
-      dropTargetKey = dropState.collection.getKeyAfter(dropTargetKey) ?? dropTargetKey;
+      let nextKey = dropState.collection.getKeyAfter(dropTargetKey);
+      if (nextKey != null) {
+        let targetLevel = dropState.collection.getItem(dropTargetKey)?.level ?? 0;
+        // Skip over any rows that are descendants of the target (can't drop an item into itself)
```

Comment:
The logic seems right here, but I don't think the comment is. `dropTargetKey` is based on the current drop target, not the dragged item. So the problem isn't that you can't drop an item on itself, but that the "after" drop position occurs after all of the children of an item, not immediately after the item itself.